src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.cs

```diff
@@ -51,12 +52,15 @@ protected override void TelemetrySessionInitialized()
                     return;
                 }
 
-                var settings = SerializeCurrentSessionSettings();
-                Contract.ThrowIfNull(settings);
+                Contract.ThrowIfNull(CurrentSession);
+
+                var settings = CurrentSession.SerializeSettings();
+                var sharedProperties = ImmutableDictionary.CreateBuilder<string, string>();
+                sharedProperties.Add("VS.Core.Version", (string)CurrentSession.GetSharedPropertyAsObject("VS.Core.ExeVersion"));
```

Comment:
💭 Interesting how this is obtained. It would have been nice to see consistent use of [`__VSSPROPID5.VSSPROPID_ReleaseVersion`](https://github.com/microsoft/vs-extension-testing/blob/a3849b605b697ddb99aa1a6106b9fa68781525dd/src/Microsoft.VisualStudio.Extensibility.Testing.SourceGenerator.UnitTests/Resources/TestGenerationForVS2012/ShellInProcess1.g.cs#L44-L65), but I can see how we would want the telemetry numbers to match exactly between devenv and OOP.

Reply:
> I can see how we would want the telemetry numbers to match exactly between devenv and OOP.

That was intentional on my part: if I'm reading the number being used for one part of telemetry and using it for another part, there's no risk of them getting out of sync in some way. And for all I know that ReleaseVersion could mean something else *too*, so best to just decouple.